### PR TITLE
:recycle: refactor: swap field

### PR DESCRIPTION
**Pull request by MarioBatalha**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206265310508493
  - https://app.asana.com/0/0/1206265310508500
  - https://app.asana.com/0/0/1206250953909085

### CODE REVIEW

File: views/swap/input/header-info.tsx [hunk 1]
```diff
@@ -1,41 +1,29 @@
...
-                  String((value / 100) * Number(balance))
-                )
-              }
-            >
-              {value < 100 ? `${value}%` : 'max'}
-            </Button>
-          );
-        })}
+const HeaderInfo: FC<HeaderInfoProps> = ({ label, balance }) => {
+  return (
```
Comment (git-marcopitra) [inline views/swap/input/header-info.tsx:7]: It's better to be without this `return`

File: views/swap/index.tsx [hunk 2]
```diff
@@ -33,7 +33,6 @@ const Swap: FC = () => {
         mx="auto"
         display="flex"
         borderRadius="2rem"
```
Comment (git-marcopitra) [inline views/swap/index.tsx:35]: why not using the theme keys?

File: views/swap/index.tsx [hunk 3]
```diff
@@ -43,13 +42,15 @@ const Swap: FC = () => {
           <Box
             left="45%"
             top="-1.25rem"
-            borderRadius="full"
             position="absolute"
-            bg="lowestContainer"
+            border="7px solid"
+            borderColor="#F8F9FD"
```
Comment (KipandaJr) [inline views/swap/index.tsx:47]: check if this color dosent has on theme

File: views/swap/input/index.tsx [hunk 4]
```diff
@@ -19,18 +18,14 @@ const Input: FC<InputProps> = ({ label }) => {
...
       <Box pl="l" pt="1rem" display="flex" justifyContent="space-between">
         <SelectToken label={label} balance={balance} />
         <TextField
           pl="-1rem"
-          placeholder="000"
+          placeholder="0"
+          color="onSurface"
           textAlign="right"
           fontSize="1.375rem"
           lineHeight="1.75rem"
```
Comment (KipandaJr) [inline views/swap/input/index.tsx:31]: check if theme dosent has this values

File: views/swap/input/select-token.tsx [hunk 5]
```diff
@@ -48,40 +43,24 @@ const SelectToken: FC<SelectTokenProps> = ({ label, balance }) => {
   return (
     <Box position="relative">
       <Button
-        pr="1rem"
-        pl="0.5rem"
+        bg="highestContainer"
+        px="m"
+        py="m"
```
Comment (KipandaJr) [inline views/swap/input/select-token.tsx:48]: in this use u can use a `p="m"` instead this approach

File: views/swap/input/select-token.tsx [hunk 6]
```diff
@@ -48,40 +43,24 @@ const SelectToken: FC<SelectTokenProps> = ({ label, balance }) => {
...
+        borderRadius="xs"
         variant="tonal"
         fontSize="0.875rem"
```
Comment (git-marcopitra) [inline views/swap/input/select-token.tsx:51]: use theme keys

Comment (KipandaJr) [inline views/swap/input/select-token.tsx:51]: check this fontSize on theme

**Review by KipandaJr — CHANGES_REQUESTED** [review]

File: views/swap/index.tsx (hunk 3 above)
Comment (git-marcopitra) [inline views/swap/index.tsx:47]: color from theme please

File: views/swap/input/index.tsx [hunk 7]
```diff
@@ -43,6 +38,39 @@ const Input: FC<InputProps> = ({ label }) => {
...
+        {label === 'from' && (
+          <Box px="xl">
+            <Slider
+              initial={0}
+              max={100}
+              onChange={() => console.log('range')}
+            />
+          </Box>
+        )}
+        <Box
```
Comment (git-marcopitra) [inline views/swap/input/index.tsx:51]: remove all the code from here, submit button must to be on the high level

**Review by git-marcopitra — APPROVED** [review]

**Review by git-marcopitra — CHANGES_REQUESTED** [review]
Requesting changes

File: views/swap/input/index.tsx [hunk 8]
```diff
@@ -43,6 +38,39 @@ const Input: FC<InputProps> = ({ label }) => {
...
+            bg="container"
+            opacity="0.4"
+            color="onSurface"
+            px="l"
+            py="s"
+            borderRadius="xs"
+            variant="tonal"
+            fontSize="s"
+          >
+            Preview swap
```
Comment (git-marcopitra) [inline views/swap/input/index.tsx:70]: Remove preview button from here

File: views/swap/input/select-token.tsx [hunk 9]
```diff
@@ -48,40 +43,23 @@ const SelectToken: FC<SelectTokenProps> = ({ label, balance }) => {
...
-            width="1.5rem"
-            height="1.5rem"
-            alignItems="center"
-            position="relative"
-            borderRadius="full"
-            justifyContent="center"
-          >
-            <BNBSVG maxWidth="1.125rem" maxHeight="1.125rem" width="100%" />
-          </Box>
+          <Typography size="large" variant="label">
```
Comment (git-marcopitra) [inline views/swap/input/select-token.tsx:53]: Why it is `PrefixIcon` and not `children`?

**Review by git-marcopitra — CHANGES_REQUESTED** [review]

File: views/swap/input/index.tsx [hunk 10]
```diff
@@ -19,30 +18,41 @@ const Input: FC<InputProps> = ({ label }) => {
...
           })}
-          fieldProps={{ borderColor: 'transparent', width: '100%' }}
+          fieldProps={{
+            borderColor: 'transparent',
+            borderRadius: 'xs',
+            width: '100%',
+          }}
         />
       </Box>
+      <Box height="2rem">
```
Comment (git-marcopitra) [inline views/swap/input/index.tsx:45]: I don't recommend to use height by default

File: views/swap/input/index.tsx [hunk 11]
```diff
@@ -19,30 +18,41 @@ const Input: FC<InputProps> = ({ label }) => {
...
+        {label === 'from' && (
+          <Box px="xl">
+            <Slider
+              initial={0}
+              max={100}
+              onChange={() => console.log('range')}
```
Comment (git-marcopitra) [inline views/swap/input/index.tsx:51]: make `onChange` works like [Sui Dapp](https://github.com/interest-protocol/sui-interface)

File: views/swap/input/select-token.tsx [hunk 12]
```diff
@@ -48,40 +43,18 @@ const SelectToken: FC<SelectTokenProps> = ({ label, balance }) => {
...
-        }
       >
-        {token.symbol}
-      </Button>
-      <Typography variant="label" size="small" mt="l">
-        Balance:
-        <Typography variant="label" size="small" color="primary" as="span">
-          {balance}
+        <Typography size="large" variant="label">
+          Select token
```
Comment (git-marcopitra) [inline views/swap/input/select-token.tsx:54]: Dynamic value based on Sui Dapp

**Review by git-marcopitra — CHANGES_REQUESTED** [review]

**Review by git-marcopitra — APPROVED** [review]

**Comment by git-marcopitra** [comment]
Fix conflits

File: components/account-info/avatar.tsx [hunk 13]
```diff
@@ -1,40 +1,38 @@
...
       {withNameOrAddress && (
-        <Typography
-          variant="label"
-          size="large"
-          mr="0.5rem"
-          width="max-content"
-        >
-          {address.slice(0, 6)}…{address.slice(-4)}
-        </Typography>
+        <Box fontFamily="Proto !important">
```
Comment (KipandaJr) [inline components/account-info/avatar.tsx:33]: why !important

File: components/account-info/index.tsx [hunk 14]
```diff
@@ -1,15 +1,99 @@
...
+          <Box
+            gap="xs"
+            p="xs"
+            display="flex"
+            cursor="pointer"
+            borderRadius="full"
+            alignItems="center"
+            border="0.25rem solid"
+            px={isOpen ? 's' : 'unset'}
+            borderColor={isOpen ? '#0053DB33' : 'transparent'}
```
Comment (KipandaJr) [inline components/account-info/index.tsx:62]: use the primary color `#0053DB` from theme and add the opacity

File: components/account-info/index.tsx [hunk 15]
```diff
@@ -1,15 +1,99 @@
...
+      )}
+      <Box
+        gap="1rem"
+        p="0.25rem"
+        cursor="pointer"
+        borderRadius="full"
+        alignItems="center"
+        border="0.25rem solid"
+        transition="all 0.3s ease-in-out"
+        borderColor="rgba(0, 83, 219, 0.16)"
```
Comment (KipandaJr) [inline components/account-info/index.tsx:81]: check this color

File: components/account-info/index.tsx [hunk 16]
```diff
@@ -1,15 +1,99 @@
...
+          </Box>
+          <MenuOptions
+            isMenuOpen={isOpen}
+            handleDisconnect={handleDisconnect}
+          />
+        </Box>
+      )}
+      <Box
+        gap="1rem"
+        p="0.25rem"
```
Comment (KipandaJr) [inline components/account-info/index.tsx:75]: get on theme

File: components/account-info/mobile-menu-options.tsx [hunk 17]
```diff
@@ -22,7 +22,7 @@ const AccountInfo: FC<AccountInfoProps> = ({
 
   return (
     <Box
-      gap="m"
+      gap="1rem"
```
Comment (KipandaJr) [inline components/account-info/mobile-menu-options.tsx:25]: get on theme

File: views/components/pool-title-bar/index.tsx [hunk 18]
```diff
@@ -13,29 +13,30 @@ const PoolTitleBar: FC<PoolTitleBarProps> = ({
 }) => (
   <Box
     mx="auto"
-    py="m"
-    gap="m"
-    px="xl"
+    py="1rem"
+    gap="1rem"
+    px="1.5rem"
```
Comment (KipandaJr) [inline views/components/pool-title-bar/index.tsx:18]: get on the theme

File: views/components/pool-title-bar/index.tsx [hunk 19]
```diff
@@ -13,29 +13,30 @@ const PoolTitleBar: FC<PoolTitleBarProps> = ({
...
-    <Button isIcon variant="text" mr="xs" onClick={onBack}>
+    <Button isIcon variant="text" mr="0.5rem" onClick={onBack}>
       <ArrowLeftSVG width="1.5rem" maxWidth="1.5rem" maxHeight="1.5rem" />
     </Button>
     <Typography
       size="large"
       variant="headline"
       textAlign="center"
-      fontSize={['3xl', '3xl', '5xl', '5xl']}
+      fontSize={['1.5rem', '1.5rem', '2rem', '2rem']}
```
Comment (KipandaJr) [inline views/components/pool-title-bar/index.tsx:33]: get on the theme

File: views/components/select-token-modal/token-modal-item.tsx [hunk 20]
```diff
@@ -41,16 +34,16 @@ const TokenModalItem: FC<TokenModalItemProps> = ({
...
-      p="xl"
+      p="1rem"
       display="flex"
       color="textSoft"
       cursor="pointer"
       alignItems="center"
+      nHover={{ bg: 'rgba(0, 83, 219, 0.08)' }}
       justifyContent="space-between"
-      nHover={{ bg: `${colors.primary}14` }}
+      bg={selected ? 'rgba(0, 83, 219, 0.08)' : 'unset'}
```
Comment (KipandaJr) [inline views/components/select-token-modal/token-modal-item.tsx:44]: get this color on theme

**Review by KipandaJr — CHANGES_REQUESTED** [review]
image

**Pull request merged**